Remove build_total and add regular build

lars-petter-hauge opened the pull request:
**Task**
Update the Travis build job

This isn't completely finished, e.g. the `before_install` could be revised.

I've tried to comply with the travis recommendation (and git itself) for each stage

markusdregi commented on .libres_version:
```diff
@@ -0,0 +1 @@
+export LIBRES_VERSION=2.3.10
```
And ECL_VERSION?

lars-petter-hauge replied:
The ECL version is given from the libres repo

markusdregi replied:
I think ERT should have its own version. Otherwise, adapting to a new major version of ecl in libres would potentially break master ERT.

lars-petter-hauge replied:
From what I could see, ERT is using a small number of functions directly from libecl. The most used is util functions, e.g. `from ecl.util.util import BoolVector`. Other than that it's mostly tests that use `ecl` on its own.

I thought that overriding libres' own defined dependencies could cause issues..

lars-petter-hauge commented:
So this is shaping up. There's some issues in the git history with lots of differences where there's really no one.. I can get that fixed.. 

But what's more is that it seems we're doing python3, while setting up conda with PyQt4 forces python2.7..

markusdregi commented on .travis.yml:
```diff
@@ -45,40 +48,62 @@ addons:
+             -DCMAKE_F_CLAGS='-Werror=all'
+             -DCMAKE_CXX_FLAGS='-Werror -Wno-unused-result'
+  - make install
+  - ctest --output-on-failure $TEST_SUITE
```
I guess `TEST_SUITE` can go?

lars-petter-hauge replied:
yes, that's wrongly copied. I'll remove it

markusdregi commented:
Looks good, besides that it seems like we have the same Python 3 issues here as we had in _libecl_. We can merge this one as is, and fix it afterwards if you prefer. See https://github.com/equinor/libecl/pull/576 for more details.

lars-petter-hauge commented:
Considering the need for PyQt4 I would guess that the same fix as libecl is getting won't be enough.

So we pretty much have two python 3 issues here..

lars-petter-hauge commented:
I managed to forget about this one.. https://github.com/equinor/ert/pull/105

Ert should be running with both PyQt4 and 5..

lars-petter-hauge commented:
I can't find that the installation isn't respecting the python version. From the travis 3.7 job:

`Installing: /home/travis/build/equinor/ert/install/lib/python3.7/dist-packages/ecl/__init__.py`

`-- Found ecl.  2.4.git >= 2.1.0 in /home/travis/build/equinor/ert/install/lib/python3.7/dist-packages`
`-- Found res.  2.4.git >= 2.1.0 in /home/travis/build/equinor/ert/install/lib/python3.7/dist-packages`

I see that we explicitly set the `PYTHONPATH` to version 2.7.. but the installation runs as it should?

lars-petter-hauge commented:
Now the travis job uses pyqt4 and 5 for python2.7 and 3.7 respectively.

markusdregi commented on .travis.yml:
```diff
@@ -15,7 +15,10 @@ dist: trusty
-    - LD_LIBRARY_PATH="$(pwd)/install/lib64"
+    - INSTALL_DIR="$(pwd)/install"
+    - LD_LIBRARY_PATH="${INSTALL_DIR}/lib:${INSTALL_DIR}/lib64"
+    - PYTHONPATH="${INSTALL_DIR}/lib/python2.7/site-packages:${INSTALL_DIR}/lib64/python2.7/site-packages:$PYTHONPATH"
```
> Found ecl.  2.4.git >= 2.1.0 in /home/travis/build/equinor/ert/install/lib/python3.7/dist-packages

I'm a bit confused here, because this does not appear to be part of what we inject in the `PYTHONPATH`?

markusdregi commented:
This looks really good now! As mentioned above, the Python modules appears to be installed at reasonable locations, but I'm a bit surprised that we manage to pick them up again in the Python 3.7 build as all of our `PYTHONPATH` injections contains `2.7`.

markusdregi commented:
> Checking the install prefix

You're right! But, then the `PYTHONPATH` manipulations are nothing but misleading, and should probably just be removed...

lars-petter-hauge commented:
I agree, I’ll get it removed. There are some unused cmake flags that must be removed, then I’m happy with this

lars-petter-hauge commented:
Removed the unused cmake flags - remnants from building all of libecl, libres and ert.

Removed the pythonpath changes

When this goes through it can be merged.
